### ISSUE TITLE
[BUG] Command+v only pastes plain text on Firefox Headless

### ISSUE THREAD

**Issue by luin** [microsoft/playwright#18339]
**Context:**
- Playwright Version: 1.27.1
- Operating System: Mac
- Node.js version: 16
- Browser: Firefox Headless

**Code Snippet**

```javascript
test('firefox headless', async ({ page }) => {
  await page.goto('https://jsbin.com/rebomam');
  await page.keyboard.press(`Meta+KeyA`);
  await page.keyboard.press(`Meta+KeyC`);

  await page.evaluate(() => {
    const dataContainer = document.createElement('div');
    dataContainer.setAttribute('contenteditable', 'true');
    dataContainer.id = '_readClipboard';
    document.body.appendChild(dataContainer);
    dataContainer.focus();
  });
  await page.keyboard.press(`Meta+KeyV`);
  const element = await page.locator('#_readClipboard');
  const data = await element.innerHTML();
  expect(data).toContain('<code>code</code>');
});
```

**Describe the bug**

The test case runs successfully with Chrome (Headed/Headless), Webkit (Headed/Headless), and Firefox (Headed), but failed on Firefox (Headless).

Firefox Headless:
```shell
> npx playwright test

Running 1 test using 1 worker
  1) [firefox] › example.spec.ts:70:1 › firefox headless ===========================================

    Error: expect(received).toContain(expected) // indexOf

    Expected substring: "<code>code</code>"
    Received string:    "<br><br>code<br><br><br>"

      84 |   const element = await page.locator('#_readClipboard');
      85 |   const data = await element.innerHTML();
    > 86 |   expect(data).toContain('<code>code</code>');
         |                ^
      87 | });
      88 |
      89 | // test('homepage has Playwright in title and get started link linking to the intro page', async ({

        at /e2e/example.spec.ts:86:16


  1 failed
    [firefox] › example.spec.ts:70:1 › firefox headless ============================================
```

Firefox Headed:
```shell
> npx playwright test --headed

Running 1 test using 1 worker

  1 passed (4s)

To open last HTML report run:

  npx playwright show-report
```

**Comment by luin** [comment]
Investigated this a bit more, and seems Firefox headless doesn't support `text/html` clipboard item. An easier way to reproduce:

```js
const { firefox } = require('playwright');

(async () => {
  const browser = await firefox.launch({
    headless: process.env.HEADLESS === 'true',
    firefoxUserPrefs: {
      'dom.events.testing.asyncClipboard': true,
      'dom.events.asyncClipboard.readText': true,
      'dom.events.asyncClipboard.clipboardItem': true,
    },
  });
  const page = await browser.newPage();
  await page.goto('https://jsbin.com/rebomam');
  await page.keyboard.press(`Meta+KeyA`);
  await page.keyboard.press(`Meta+KeyC`);

  console.log(
    await page.evaluate(async () => {
      return (await navigator.clipboard.read())[0].types;
    }),
  );

  await browser.close();
})();
```

Results:

```shell
> HEADLESS=true node e2e/play.js
[ 'text/plain' ]
> HEADLESS=flsy node e2e/play.js
[ 'text/html', 'text/plain' ]
```

**Comment by mxschmitt** [comment]
Looks like this upstream issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1686012

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.

**Comment by thewilkybarkid** [comment]
Seems to have been fixed in `v1.41` (Firefox `v121.0`).
